### PR TITLE
Migrate to dev/1.0.0

### PR DESCRIPTION
**Pull request by Charles-Schleich**
Migrate API to Dev 1.0.0

### CODE REVIEW

File: zenoh-plugin-mqtt/src/lib.rs [hunk 1]
```diff
@@ -481,9 +485,15 @@ fn treat_admin_query(query: Query, admin_keyexpr_prefix: &keyexpr, config: &Conf
     // send replies
     for (ke, v) in kvs.drain(..) {
         let admin_keyexpr = admin_keyexpr_prefix / ke;
-        use zenoh::prelude::sync::SyncResolve;
-        if let Err(e) = query.reply(Ok(Sample::new(admin_keyexpr, v))).res_sync() {
-            tracing::warn!("Error replying to admin query {:?}: {}", query, e);
+        match zenoh::bytes::Serialize::serialize(ZSerde, v) {
```
Comment (milyin) [inline zenoh-plugin-mqtt/src/lib.rs:488]: No need to use ZSerde directly, here `try_into()` works. See https://github.com/eclipse-zenoh/zenoh-plugin-mqtt/pull/131

File: zenoh-plugin-mqtt/src/lib.rs [hunk 2]
```diff
@@ -134,15 +136,17 @@ async fn run(
...
+
+    let admin_keyexpr_expr = (&admin_keyexpr_prefix) / (ke_for_sure!("**"));
     tracing::debug!("Declare admin space on {}", admin_keyexpr_expr);
     let config2 = config.clone();
     let _admin_queryable = zsession
         .declare_queryable(admin_keyexpr_expr)
         .callback(move |query| treat_admin_query(query, &admin_keyexpr_prefix, &config2))
-        .res()
-        .await
+        .wait()
```
Comment (milyin) [inline zenoh-plugin-mqtt/src/lib.rs:149]: wait() makes this code synchronous, it should be replaced to just .await

File: zenoh-plugin-mqtt/src/config.rs [hunk 3]
```diff
@@ -94,6 +97,10 @@ where
     deserializer.deserialize_option(OptPathVisitor)
 }
 
+fn default_mqtt_tx_channel_size() -> usize {
```
Comment (milyin) [inline zenoh-plugin-mqtt/src/config.rs:100]: addind new functionality should be in another PR
I tinnk the https://github.com/eclipse-zenoh/zenoh-plugin-mqtt/pull/131 should be applied first and this `mqtt_tx_channel_size` update  should be applied after it, better with reference to the corresponding issue

File: Cargo.toml [hunk 4]
```diff
@@ -46,21 +46,25 @@ rustls-pemfile = "2.0.0"
...
-zenoh = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main", features = [
+zenoh = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "dev/1.0.0", features = [
   "plugins",
   "unstable",
+  "internal",
 ] }
-zenoh-collections = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh-core = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main" }
-zenoh-ext = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "main", features = [
+zenoh-collections = { version = "0.11.0-dev", git = "https://github.com/eclipse-zenoh/zenoh.git", branch = "dev/1.0.0" }
```
Comment (milyin) [inline Cargo.toml:56]: internal crates are not needed now, only `zenoh` is enough.
For plugins `zenoh-plugin-trait` is necessary also because of macro exports from it and that's all

**Review by milyin — CHANGES_REQUESTED** [review]

**Comment by milyin** [comment]
I've made update https://github.com/eclipse-zenoh/zenoh-plugin-mqtt/pull/131 based on these changes, but without adding new functionality

File: zenoh-plugin-mqtt/src/lib.rs (hunk 1 above)
Comment (Charles-Schleich) [reply zenoh-plugin-mqtt/src/lib.rs:488]: Sure thing.

File: zenoh-plugin-mqtt/src/lib.rs (hunk 2 above)
Comment (Charles-Schleich) [reply zenoh-plugin-mqtt/src/lib.rs:149]: Ah i see, i didnt realize it made the code Sync,

File: Cargo.toml (hunk 4 above)
Comment (Charles-Schleich) [reply Cargo.toml:56]: Perfect. https://github.com/eclipse-zenoh/zenoh-plugin-mqtt/pull/131 first then we can merge this branch as an update after the fact.

**Comment by Mallets** [comment]
@Charles-Schleich #131 has been merged, please update this PR accordingly
